kinesis/kinesis.go

```diff
@@ -284,27 +293,34 @@ func (outputPlugin *OutputPlugin) sendCurrentBatch(records *[]*kinesis.PutRecord
 // processAPIResponse processes the successful and failed records
 // it returns an error iff no records succeeded (i.e.) no progress has been made
 func (outputPlugin *OutputPlugin) processAPIResponse(records *[]*kinesis.PutRecordsRequestEntry, dataLength *int, response *kinesis.PutRecordsOutput) (int, error) {
+
+	var retCode int = fluentbit.FLB_OK
+
 	if aws.Int64Value(response.FailedRecordCount) > 0 {
 		// start timer if all records failed (no progress has been made)
 		if aws.Int64Value(response.FailedRecordCount) == int64(len(*records)) {
 			outputPlugin.timer.Start()
 			return fluentbit.FLB_RETRY, fmt.Errorf("PutRecords request returned with no records successfully recieved")
 		}
 
-		logrus.Warnf("[kinesis %d] %d records failed to be delivered. Will retry.\n", outputPlugin.PluginID, aws.Int64Value(response.FailedRecordCount))
+		logrus.Warnf("[kinesis %d] %d/%d records failed to be delivered. Will retry.\n", outputPlugin.PluginID, aws.Int64Value(response.FailedRecordCount), len(*records))
 		failedRecords := make([]*kinesis.PutRecordsRequestEntry, 0, aws.Int64Value(response.FailedRecordCount))
 		// try to resend failed records
 		for i, record := range response.Records {
 			if record.ErrorMessage != nil {
 				logrus.Debugf("[kinesis %d] Record failed to send with error: %s\n", outputPlugin.PluginID, aws.StringValue(record.ErrorMessage))
 				failedRecords = append(failedRecords, (*records)[i])
 			}
+
 			if aws.StringValue(record.ErrorCode) == kinesis.ErrCodeProvisionedThroughputExceededException {
-				logrus.Warnf("[kinesis %d] Throughput limits for the stream may have been exceeded.", outputPlugin.PluginID)
-				return fluentbit.FLB_RETRY, nil
+				retCode = fluentbit.FLB_RETRY
 			}
 		}
 
+		if retCode == fluentbit.FLB_RETRY {
+			logrus.Warnf("[kinesis %d] Throughput limits for the stream may have been exceeded.", outputPlugin.PluginID)
```

Comment:
small nit: I'm fine with this, but it'd be best to have a variable in the function called something like `limitsExceeded` and set it to true on line 316, and then print the warning here if it is true. 

Checking `if retCode == fluentbit.FLB_RETRY` means that if some day someone carelessly updates the plugin to set the retCode to retry for a different condition, this warning will be printed in error.